packages/wallet/src/KeyManagement/util/stubSignTransaction.ts

```diff
@@ -9,14 +9,19 @@ const randomPublicKey = () => Cardano.Ed25519PublicKey(Array.from({ length: 64 }
 export const stubSignTransaction = async (
   txBody: Cardano.NewTxBodyAlonzo,
   knownAddresses: GroupedAddress[],
-  inputResolver: InputResolver
-): Promise<Cardano.Signatures> =>
-  new Map(
-    (await ownSignatureKeyPaths(txBody, knownAddresses, inputResolver)).map(() => [
-      randomPublicKey(),
-      Cardano.Ed25519Signature(
-        // eslint-disable-next-line max-len
-        'ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
-      )
-    ])
+  inputResolver: InputResolver,
+  signOptions?: SignTransactionOptions
+): Promise<Cardano.Signatures> => {
+  const mockSignature = Cardano.Ed25519Signature(
+    // eslint-disable-next-line max-len
+    'ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
   );
+
+  const signatureKeyPaths = await ownSignatureKeyPaths(txBody, knownAddresses, inputResolver);
+  const totalSignature = signatureKeyPaths.length + (signOptions?.extraSigners?.length || 0);
```

Comment:
This is a problem because extraSigners can return more than 1 signature, but we also don't want to call it here so there's no way of knowing how many signatures we'll have. Perhaps let's change `SignTx` to return just 1 signature? Then if needed you could create a util like `keyAgentAsSigner(keyAgent: KeyAgent): SignTx`

Reply:
addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/404/commits/1745bf74c984767cac4ca69730a1da8f2ee40a90